eclair-core/src/main/scala/fr/acinq/eclair/message/Postman.scala

```diff
@@ -88,10 +88,10 @@ object Postman {
           Behaviors.same
         case SendMessage(intermediateNodes, destination, replyPath, messageContent, replyTo, timeout) =>
           val messageId = randomBytes32()
-          val replyRoute = replyPath.map(replyHops => {
+          val replyRoute = replyPath.flatMap(replyHops => {
```

Comment:
We want `replyRoute` to be `None` if and only if `replyPath` is `None` too. You're breaking the API by doing that.

Reply:
You're talking about the case where `buildRoute` would return a failure, right? That case was previously simply unhandled and would have crashed the actor without sending a response to the `replyTo` (because the `.get` would have failed), which is strictly worse, isn't it?

But you're raising a good point, I assumed that the reply path would never actually _start_ at our node. It would feel dumb, because even if you want the receiver to directly send a response to you, the reply path you'd use would be a blinded path `Receiver -> You` so that they're never 100% sure you sent the message, right? Should we check that though and reject reply paths that start at our node?

Reply:
This change disappeared in https://github.com/ACINQ/eclair/pull/2589/commits/9dd1f2e822a4563cefd66618326da537cdf2b612